src/__tests__/indexing.ts

```diff
@@ -0,0 +1,292 @@
+          expect(retrievedCreatedDoc1.id.toString()).toEqual(doc1.id.toString())
+          expect(retrievedCreatedDoc1.content).toEqual(doc1.content)
+          expect(StreamUtils.serializeState(retrievedCreatedDoc1.state)).toEqual(
+            StreamUtils.serializeState(doc1.state)
+          )
+
+          await doc1.replace({ data: Math.floor(Math.random() * 1000) }, { anchor: false })
+          const doc2 = await ModelInstanceDocument.create(
+            ceramicInstance,
+            { data: Math.floor(Math.random() * 1000) },
+            { model: TEST_MODEL },
+            { anchor: false }
+          )
+
+          await TestUtils.delay(5 * 1000)
+
+          const resultsAfterReplace = await ceramicInstance.index
+            .query({ model: TEST_MODEL, last: 10 })
+            .then(resultObj => extractDocuments(ceramicInstance, resultObj))
+          expect(resultsAfterReplace.length).toBeGreaterThanOrEqual(2)
+
+          const retrievedReplacedDoc1 = resultsAfterReplace.at(-2) as ModelInstanceDocument
+          expect(retrievedReplacedDoc1.id.toString()).toEqual(doc1.id.toString())
+          expect(retrievedReplacedDoc1.content).toEqual(doc1.content)
+          expect(StreamUtils.serializeState(retrievedReplacedDoc1.state)).toEqual(
+            StreamUtils.serializeState(doc1.state)
+          )
+
+          const retrievedDoc2 = resultsAfterReplace.at(-1) as ModelInstanceDocument
+          expect(retrievedDoc2.id.toString()).toEqual(doc2.id.toString())
+          expect(retrievedDoc2.content).toEqual(doc2.content)
+          expect(StreamUtils.serializeState(retrievedDoc2.state)).toEqual(
+            StreamUtils.serializeState(doc2.state)
+          )
+        }
+      )
+
+      test.each(singleNodeTestCases)(
+        'Can filter by DID -- %s',
+        async (_, ceramicInstance: CeramicApi) => {
+          const did1 = originalDid
+          const did2 = await createDid()
+
+          const doc1 = await ModelInstanceDocument.create(
+            ceramicInstance,
+            { data: Math.floor(Math.random() * 1000) },
+            { model: TEST_MODEL, controller: did1.id },
+            { anchor: false }
+          )
+
+          ceramicInstance.did = did2
+          const doc2 = await ModelInstanceDocument.create(
+            ceramicInstance,
+            { data: Math.floor(Math.random() * 1000) },
+            { model: TEST_MODEL, controller: did2.id },
+            { anchor: false }
+          )
+
+          await TestUtils.delay(5 * 1000)
+
+          const did1Results = await ceramicInstance.index
+            .query({ model: TEST_MODEL, last: 10, account: did1.id })
+            .then(resultObj => extractDocuments(ceramicInstance, resultObj))
+            .catch(err => {
+              throw new Error(
+                `There was a problem retrieving the MID for model ${TEST_MODEL}: ${err}`
+              )
+            })
+
+          expect(did1Results.length).toBeGreaterThanOrEqual(1)
```

Comment:
should we also test that the second to last doc **isn't** the other doc we created in this test, to ensure that we're appropriately filtering it out?